syntax = "proto3";

package hitszids.wf.opendlp.api.v1;

option java_multiple_files = true;
option java_outer_classname = "OpenDlpProto";
option java_package = "org.hitszids.wf.opendlp.api.v1";

service SensitiveDataAnalyzerService{
  // 敏感数据识别推理服务
  rpc SensitiveAnalyze(SensitiveRequest) returns (SensitiveResponse) {}
}

// 敏感数据识别服务请求
message SensitiveRequest{
  // 待预测文件路径，目前仅支持csv文件
  string to_analyze_file_path = 1;

  // 自定义正则json
  string user_define_pattern_file = 2;

  // 敏感数据列类型判断阈值,json串{敏感数据类型:阈值}
  string thresholds = 3;
}

// 敏感数据识别服务响应
message SensitiveResponse{
  // 状态码
  Status status = 1;

  // 识别结果。json串{列名:{'success':bool, 'type':敏感数据类型, 'percent':'识别出的数量/列中数据总量（最多1000）'}}
  string result = 2;
}

// 接口请求的应答状态码、消息
message Status {
    // 应答状态码
    StatusCode code = 1;

    // 应答消息
    string msg = 2;
}

// 应答状态码枚举值
enum StatusCode {
    // 成功,所有列都正常返回识别结果
    OK = 0;

    // 参数有误, 文件不存在也属于参数有误
    PARAMETER_ERROR = 10000;

    // 读取数据文件失败
    FILE_READ_ERROR = 10001;

    // json文件解析失败
    JSON_FILE_PARSE_ERROR = 10002;

    // 存在某列进行敏感数据识别时失败
    ANY_COLUMN_RECOGNIZE_ERROR = 10003;
}

